composer/text-composer.html

```diff
@@ -232,15 +241,47 @@
           if (event.ctrlKey || event.shiftKey || event.altKey || event.metaKey) {
             return;
           }
+          let range = this.$.text.selectionRange;
+          // if text is selected - return i.e. default behavior
+          if (!(range && range.collapsed)) {
+            return;
+          }
+
           let value = this.$.text.value || "";
           let hasList = (value.indexOf("<ol") >= 0) || (value.indexOf("<ul") >= 0);
           if (!hasList) {
             let text = this.$.text.textValue.trim();
             if (!text) {
               return;
             }
+
+            // select from current cursor to end of text
+            let sOffset = range ? (range.startOffset || 0) : 0;
+            let sContainer = range.startContainer;
+            let cutText = false;
+            let cutContent = "";
+            try {
+              this.selectEnd();
+              range = this.$.text.selectionRange;
+              range.setStart(sContainer, sOffset);
+              let s = this.$.text.selection;
+              s.removeAllRanges();
+              s.addRange(range);
+              if (!range.collapsed) {
+                let clonedSelection = range.cloneContents() || "";
+                let div = document.createElement('div');
+                div.appendChild(clonedSelection);
+                cutContent = div.innerHTML;
+                this.$.text.cutSelection();
+                cutText = true;
```

Comment:
Some browsers (Chrome) does not let you call paste programmatically for security reasons, so here I try to extract the html fragment of the text after the cursor. This is used to replace the inner html if paste fails. 
Since text could be html with many nodes you have to add the fragment inside a dummy div to get raw html out.

Reply:
Does this mean that formatting (bold, links, etc.) in the tail of the paragraph is lost when the paragraph gets split?

Reply:
No. this is precisely to avoid that from happening.